### ISSUE TITLE
"ember install package@version" fails after npm

### ISSUE THREAD

**Issue by mike-north** [ember-cli/ember-cli#5880]
### What went wrong

I (and I think most users) expect `ember install` to match reasonably with `npm install` in terms of support for the way packages are defined. For example, I would expect both of these to result in the same version of the package being added to my project:

```sh
npm install ember-cp-validations@~2.7.0
ember install ember-cp-validations@~2.7.0
```

I'm now seeing that what happens if you attempt the `ember install` above is that the modification is made to `package.json`, but then the user gets an error

```
Install failed. Could not find addon with name: ember-cp-validations@~2.7.0
```

### Suspected cause

To me, this smells like we're losing the distinction between name and "name@target" at the time we attempt to run an installation blueprint. Sure enough, even after you get the above error, if you run the blueprint explicitly

```
ember g ember-cp-validations
```

Everything is run as you would expect.

### This used to work fine

- I tested with ember-cli `1.13.8` and `2.5.0`, and this bug doesn't appear to be present

### To reproduce

```sh
ember new ember123 && \
   cd ember123 && \
   npm install --save-dev ember-cli@2.6.0-beta.2 && \
   ember install ember-cp-validations@~2.7.0
```
 
Output from ember version --verbose:
```
ember-cli: 2.6.0-beta.2
http_parser: 2.6.2
node: 5.10.1
v8: 4.6.85.31
uv: 1.8.0
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 56.1
modules: 47
openssl: 1.0.2g
os: darwin x64
```

**Comment by Turbo87** [comment]
this might be related to https://github.com/ember-cli/ember-cli/pull/5752

**Comment by mike-north** [comment]
very likely. For any others coming across this issue, it affects `this.addAddonToProject` as well, and thus is affecting some addons with nested other addons.

**Comment by ooluwadiya** [comment]
The same issue occurs with installation from .tgz archive files e.g. https://<hostname>:<port>/<path>/addon-1.2.3.tgz

**Comment by mike-north** [comment]
- [ ] @martndemus can you make sure we have a test case in place for tarball URLs in #5883

**Comment by Turbo87** [comment]
Resolved by #5963
